src/Datadog.Trace.ClrProfiler.Managed/Emit/ModuleLookup.cs

```diff
@@ -21,6 +22,11 @@ internal static class ModuleLookup
         private static int _failures = 0;
         private static bool _shortCircuitLogicHasLogged = false;
 
+        public static Module GetByPointer(long moduleVersionPointer)
+        {
+            return Get(PointerHelpers.GetGuidFromNativePointer(moduleVersionPointer));
```

Comment:
Although the code does not appear in this PR, the `GetGuidFromNativePointer` method can throw an `OverflowException` in 32-bit environments. It's very unlikely to bite us but given that you're making use of this method, I would suggest putting a `try catch` around `var ptr = new IntPtr(nativePointer)` in `PointerHelpers.GetGuidFromNativePointer`.

Reply:
I'm fine with doing that, but we are in a pretty bad spot if this ever fails to resolve.

Reply:
Yeah but we're running inside another app -- we need to be diligent about even the smallest possible exception.

Reply:
I'm happy to log detail, but this is kind of catastrophic if that lookup doesn't work.

Reply:
We shouldn't crash the app though. That is more catastrophic. Catching the exception and logging and we can attempt to degrade gracefully.

Reply:
What @colin-higgins is trying to say is that we _can't_ degrade gracefully at this point. If we can't resolve the original method and call it, we are already going to break the app and there's no way back. I think the best we can do is throw an exception with a helpful message (and log it). The only other alternative is throwing the `OverflowException` without useful context.

Reply:
So, the wrapper method has been called, it tries to get the module of the original method, and that fails. That's the scenario, right? I wonder in that case if we should bail out of this method after logging the error. Not guaranteed that the app will crash I don't think. It might.

Reply:
Resolving this, I'll handle tracking in a follow up.